core/state/statedb.go

```diff
@@ -221,6 +221,13 @@ func (s *StateDB) EnablePipeCommit() {
 	}
 }
 
+func (s *StateDB) IsPipeCommit() bool {
+	if s.snap != nil {
```

Comment:
I think `return s.pipeCommit` will be good enough? Because only when snap is not nil, pipecommit will be true when we do `EnablePipeCommit`

Reply:
you are right